include/shuffle/PhiloxShuffle.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+#include "shuffle/BijectiveFunctionScanShuffle.h"
+
+__host__ __device__ uint32_t mulhilo(uint64_t a, uint32_t b, uint32_t &hip) {
+  uint64_t product = a * uint64_t(b);
+  hip = product >> 32;
+  return uint32_t(product);
```

Comment:
Should we use the mulhi/low intrinsics like wyhash does?

Reply:
Note that I believe our constant should be 32 bits as well. The description of the algorithm assumes the constant is the same length as the value multiplied

Reply:
I briefly played with cuda intrinsics, without any significant impact on benchmarks. It could probably be optimised but I don't think it's worth it.

Reply:
Fair enough. I think using 64 bit should be fine as well since crush resistance should transfer to a subset of bits. Though it may be a problem that we dont use the upper 64 bits, in which case we should use one of the 32 bit constants